Mask secrets in CommandInvokedEvent and Command logger

OndroMih opened the pull request:
This fixes a security issue in the Command Logger feature, which allowed to expose passwords used in some admin commands. This fix replaces passwords in the logged messages by "******".

OndroMih committed:
Mask secrets in CommandInvokedEvent and Command logger

pzygielo commented on nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/CommandRunnerImpl.java:
```diff
@@ -1642,12 +1643,22 @@ public void executeFromCheckpoint(JobManager.Checkpoint checkpoint, boolean reve
                 .publish(event);
     }
 
+    private ParameterMap maskSecretParameters(ParameterMap parameters) {
```
Could it be `static` and have unit test?

OndroMih replied:
OK. I will add this specific case into the CommandLoggerTest tests.

pzygielo commented on nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/CommandRunnerImpl.java:
```diff
@@ -1642,12 +1643,22 @@ public void executeFromCheckpoint(JobManager.Checkpoint checkpoint, boolean reve
         final CommandInvokedEvent event = new CommandInvokedEvent(
                 invocation.name(),
-                invocation.parameters(),
+                maskSecretParameters(invocation.parameters()),
```
The current failures suggest (to me), that `invocation.parameters()` can be `null`.

OndroMih committed:
Mask secrets - tests, refactoring, null check

Extracted ExecutionContext inner class to package-private class CommandRunnerExecutionContext. Was too big and deserved a file on its own.
Moved mask method to ParameterMap and added unit tests.

OndroMih committed:
Mask secrets - fix for Java 11, another test

A test to verfiy that the command logger doesn't log passwords

hs536 reviewed (approved):
LGTM
